_source/_posts/2022-08-30-introducing-the-new-okta-mobile-sdks.md

```diff
@@ -0,0 +1,333 @@
+
+```swift
+// Create the flow, using the Okta.plist configuration
+let flow = TokenExchangeFlow()
+
+// Exchange the ID and Device tokens for access tokens.
+let token = try await flow.start(with: [
+    .actor(type: .deviceSecret, value: deviceToken),
+    .subject(type: .idToken, value: idToken)
+])
+
+// Save the user's token
+let credential = try Credential.store(token)
+```
+
+Once you've authenticated a user with the `device_sso` scope, Okta will return both an ID Token and Device Secret in the token response. If you save this in a secure location accessible to all your apps (for example, in a Keychain within a shared App Group), you can supply those strings to the TokenExchangeFlow class.
+
+### Improved token storage and biometric support
+
+Like the Okta OIDC SDK, the new Swift SDK uses the Apple Keychain to securely store tokens. With the new [Credential.store](https://okta.github.io/okta-mobile-swift/development/authfoundation/documentation/authfoundation/credential/store(_:tags:security:)) function however, more options are available to you at the time you store a token. Security settings or developer-assigned tags can optionally be supplied to customize how those tokens will be stored.
+
+```swift
+// Accept defaults
+try Credential.store(token)
+
+// Assign tags to later retrieve specific tokens
+try Credential.store(token, tags: ["tag": "value"])
+
+// Share the token between an app's extensions
+try Credential.store(token, security: [
+    .accessGroup("com.myapp.access-group")
+])
+
+// Secure the token with biometrics, and require
+// the device to be unlocked
+try Credential.store(token, security: [
+    .accessibility(.unlocked),
+    .accessControl(.biometryAny)
+])
+```
+
+The tags option can be used to help manage the tokens stored in the keychain, which helps when your app works with multiple tokens or accounts. This is a feature I'm very excited about, due to the advanced use cases it unlocks. Let's dig into that in the next section.
+
+#### Supporting multiple tokens / users
+
+We now support storing multiple tokens or user accounts in a single application. This is a big improvement over the legacy Okta OIDC SDK, which made the assumption that an app would only ever have a single user at a time. Storing multiple tokens helps you create more sophisticated apps, such as:
+
+* Multi-user applications that allow users to quick switch between profiles
+* Applications that utilize app extensions, or widgets that wish to use separate tokens for each extension, for security reasons
+* Apps that use separate tokens with different scopes for escalated access
+
+For a host of reasons, the AuthFoundation library now supports the storage and use of multiple tokens simultaneously, while still supporting traditional single-user applications via the Credential.default property.
```

Comment:
```suggestion
For a host of reasons, the AuthFoundation library now supports the storage and use of multiple tokens simultaneously, while still supporting traditional single-user applications via the `Credential.default` property.
```
adding "code" tag here

Reply:
I think I missed that one. It should've been a link to the appropriate docs. https://okta.github.io/okta-mobile-swift/development/authfoundation/documentation/authfoundation/credential/default

Reply:
Updated now, thanks for catching